src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyRelayedResponse.java

```diff
@@ -126,13 +126,16 @@ private void onResponseBegin(
         final long affinity = begin.affinity();
         final OctetsFW extension = begin.extension();
 
-        final HttpBeginExFW httpBeginEx = extension.get(factory.httpBeginExRO::wrap);
-        final boolean hasEtag = httpBeginEx.headers().matchFirst(h -> ETAG.equals(h.name().asString())) != null;
-        final HttpHeaderFW status = httpBeginEx.headers().matchFirst(h -> STATUS.equals(h.name().asString()));
+
+        HttpBeginExFW httpBeginEx = extension.get(factory.httpBeginExRO::tryWrap);
+        final HttpBeginExFW httpBeginExFinal = (httpBeginEx == null) ? factory.defaultHttpBeginExRO : httpBeginEx;
```

Comment:
We usually call these `httpBeginEx0` to handle `final` requirement, say for lambdas.